src/relayer/publicDecrypt.ts

```diff
@@ -184,7 +290,20 @@ export const publicDecryptRequest =
       throw Error('KMS signers threshold is not reached');
     }
 
-    const results = deserializeDecryptedResult(handles, decryptedResult);
+    const clearValues: ClearValues = deserializeClearValues(
+      handles,
+      decryptedResult,
+    );
+
+    const abiEnc = abiEncodeClearValues(clearValues);
+    const decryptionProof = buildDecryptionProof(
+      kmsSignatures,
+      signedExtraData,
+    );
 
-    return results;
+    return {
+      clearValues,
+      abiEncodedClearValues: abiEnc.abiEncodedClearValues,
```

Comment:
I have a question where you decoded and re-encoded abi string.

Since the tests are passing, I assume this should be correct, but could you provide a bit of context on why we need this ? Because, when I check [how we did it on relayer side](https://github.com/zama-ai/console/blob/4a16aab94e1e69f4298bdb532bf4571942ed823e/apps/relayer/src/blockchain/fhevm/ethereum/transaction_calldata.rs#L15), previously used in making the oracle callback, decryption proof was passed as is.